### PR TITLE
Set of thresholds in SimpleCnec

### PR DESCRIPTION
**Pull request by terriervik**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** *(You can also link to an open issue here)*
A SimpleCnec contains only one AbstractThreshold.

**What is the new behavior (if this is a feature change)?**
A SimpleCnec contains a set of AbstractThreshold.

### CODE REVIEW

File: data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java [hunk 1]
```diff
@@ -27,24 +28,25 @@
...
-                      @JsonProperty("thresholds") AbstractThreshold thresholds, @JsonProperty("state") State state) {
+                      @JsonProperty("thresholds") Set<AbstractThreshold> thresholds, @JsonProperty("state") State state) {
         super(id, name);
 
         this.networkElement = networkElement;
-        this.thresholds = thresholds.copy();
-        this.thresholds.setNetworkElement(networkElement);
+        this.thresholds = new HashSet<>();
+        thresholds.forEach(threshold -> this.thresholds.add(threshold.copy()));
+        this.thresholds.forEach(threshold -> threshold.setNetworkElement(networkElement));
```
Comment (TheMaskedTurtle) [inline data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java]: You can iterate only once

File: data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java [hunk 2]
```diff
@@ -53,31 +55,33 @@ public NetworkElement getNetworkElement() {
...
         return thresholds;
     }
 
-    public void setThresholds(AbstractThreshold thresholds) {
+    public void setThresholds(Set<AbstractThreshold> thresholds) {
         this.thresholds = thresholds;
     }
 
+    public void addThreshold(AbstractThreshold threshold) {
+        thresholds.add(threshold);
```
Comment (TheMaskedTurtle) [inline data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java]: you should set network element of the threshold here

Comment (terriervik) [reply data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java]: sure

File: data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java (hunk 1 above)
Comment (terriervik) [reply data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java]: True!

File: data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java [hunk 3]
```diff
@@ -173,7 +188,8 @@ public boolean equals(Object o) {
         }
         SimpleCnec cnec = (SimpleCnec) o;
         return super.equals(cnec) && networkElement.equals(cnec.getNetworkElement())
-            && state.equals(cnec.getState()) && thresholds.equals(cnec.thresholds);
+            && state.equals(cnec.getState())
+            && thresholds.equals(cnec.thresholds);
```
[the remark this answers is not in the document]
Comment (terriervik) [reply data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCnec.java]: So nice for them!

**Pull request merged**